lib/ansible/galaxy/api.py

```diff
@@ -555,16 +675,41 @@ def get_collection_versions(self, namespace, name):
         if 'v3' in self.available_api_versions:
             api_path = self.available_api_versions['v3']
             pagination_path = ['links', 'next']
+            version_key = 'highest_version'
             relative_link = True  # AH pagination results are relative an not an absolute URI.
         else:
             api_path = self.available_api_versions['v2']
             pagination_path = ['next']
+            version_key = 'latest_version'
+
+        collection_info_url = _urljoin(self.api_server, api_path, 'collections', namespace, name, '/')
+        versions_url = _urljoin(self.api_server, api_path, 'collections', namespace, name, 'versions', '/')
+        versions_url_info = urlparse(versions_url)
+
+        # We should only rely on the cache if the latest version has not changed. This may slow things down but it
+        # ensures we are not waiting a day before finding any new collections that have been published.
+        # This does run the risk that an older version was published after a newer version but that will be invalidated
```

Comment:
I'm not sure if  this would really be a problem since this code path is used for unspecified versions (which should download the max version, rather than last published). Did you consider using the `modified` field instead of highest_version/latest_version? Or since versions are additive, maybe we could use the published version count from the versions_url.

Reply:
Good idea on checking the count or `modified` field, will have to double check on all 3 API endpoints to see if they all provide a universal field I can get that information from.